purchases_test.go

```diff
@@ -0,0 +1,259 @@
+package recurly_test
+
+import (
+	"bytes"
+	"encoding/xml"
+	"fmt"
+	"net/http"
+	"strconv"
+	"testing"
+
+	"github.com/blacklightcms/recurly"
+)
+
+func TestPurchases_Purchase_Encoding(t *testing.T) {
+	tests := []struct {
+		v        recurly.Purchase
+		expected string
+	}{
+		{
+			expected: "<purchase><account></account><adjustments></adjustments><currency></currency><gift_card></gift_card><coupon_codes></coupon_codes><subscriptions></subscriptions></purchase>",
+		},
+		{
+			// From Recurly examples: https://dev.recurly.com/docs/create-purchase
+			v: recurly.Purchase{
+				CollectionMethod:      "automatic",
+				Currency:              "USD",
+				CustomerNotes:         "Some notes for the customer.",
+				TermsAndConditions:    "Our company terms and conditions.",
+				VATReverseChargeNotes: "Vat reverse charge notes.",
+				Account: recurly.Account{
+					Code: "c442b36c-c64f-41d7-b8e1-9c04e7a6ff82",
+					// ShippingAddresses: []recurly.ShippingAddress{},
+					BillingInfo: &recurly.Billing{
+						Address:   "400 Alabama St",
+						City:      "San Francisco",
+						Country:   "US",
+						FirstName: "Benjamin",
+						LastName:  "Du Monde",
+						Month:     12,
+						Number:    4111111111111111,
+						State:     "CA",
+						Year:      2019,
+						Zip:       "94110",
+					},
+				},
+				Adjustments: []recurly.Adjustment{
+					recurly.Adjustment{
+						ProductCode: "4549449c-5870-4845-b672-1d07f15e87dd",
+						Quantity:    1,
+						// RevenueScheduleType: "at_invoice",
+						UnitAmountInCents: 1000,
+						Description:       "Description of this adjustment",
+					},
+				},
+				Subscriptions: []recurly.NewSubscription{
+					recurly.NewSubscription{PlanCode: "plan1"},
+				},
+				CouponCodes: []string{"coupon1", "coupon2"},
+				GiftCard:    "ABC1234",
+			},
+			expected: "<purchase>" +
+				"<account><account_code>c442b36c-c64f-41d7-b8e1-9c04e7a6ff82</account_code>" +
+				// TODO: ShippingAddresses not yet modeled
```

Comment:
There's a PR up for shipping addresses. We'll merge that in before this one so you can include it here.

Reply:
Heh. Yah, Eric and I coworkers... He just got his in before I did. 😁

Reply:
I think adding ShippingAddresses here is the last piece and then we can merge.